packages/devextreme/js/__internal/core/widget/dom_component.ts

```diff
@@ -142,9 +142,18 @@ class DOMComponent<
 
   // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
   _setOptionsByDevice(instanceCustomRules): void {
-    // @ts-expect-error
-    // eslint-disable-next-line @stylistic/max-len
-    super._setOptionsByDevice([].concat(this.constructor._classCustomRules || [], instanceCustomRules || []));
+    const ctor = this.constructor;
+
+    const hasOwnCustomRules = Object.prototype.hasOwnProperty.call(ctor, '_classCustomRules');
+    const hasOwnDefaultOptions = Object.prototype.hasOwnProperty.call(ctor, 'defaultOptions');
```

Comment:
Why we need to check for having own `defaultOptions`?
When can we or customers overwrite it?